From: Christoph Hellwig <hch@lst.de>
Date: Wed, 23 Aug 2017 19:10:28 +0200
Subject: [PATCH] raid5: remove a call to get_start_sect
Git-commit: 10433d04b8e647a50feffec72fd3cf40ce42b084
Patch-Mainline: v4.14-rc1
References: FATE#323952, FATE#322506

The block layer always remaps partitions before calling into the
->make_request methods of drivers.  Thus the call to get_start_sect in
in_chunk_boundary will always return 0 and can be removed.

Reviewed-by: Shaohua Li <shli@fb.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Jens Axboe <axboe@kernel.dk>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/md/raid5.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/md/raid5.c b/drivers/md/raid5.c
index e874136c..9ea41cd 100644
--- a/drivers/md/raid5.c
+++ b/drivers/md/raid5.c
@@ -5099,7 +5099,7 @@ static int raid5_congested(struct mddev *mddev, int bits)
 static int in_chunk_boundary(struct mddev *mddev, struct bio *bio)
 {
 	struct r5conf *conf = mddev->private;
-	sector_t sector = bio->bi_iter.bi_sector + get_start_sect(bio->bi_bdev);
+	sector_t sector = bio->bi_iter.bi_sector;
 	unsigned int chunk_sectors;
 	unsigned int bio_sectors = bio_sectors(bio);
 
-- 
1.8.5.6

